# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1463449870 14400
#      Mon May 16 21:51:10 2016 -0400
# Node ID fccffef9d210af369031c64da340da9fa5230f79
# Parent  db71f8a7d8963dcef3da9512611cb1c9e3f820f1
attempt to ensure that oct-conf-post.h is built early and as needed

* Makefile.am (GEN_CONFIG_INC): New variable.
($(GEN_CONFIG_INC) : %.h : %.in.h config.status): New pattern rule.
(BUILT_SOURCES): Include oct-conf-post.h in the list.

diff --git a/Makefile.am b/Makefile.am
--- a/Makefile.am
+++ b/Makefile.am
@@ -74,16 +74,22 @@ GEN_CONFIG_SHELL = \
   build-aux/subst-cross-config-vals.sh \
   build-aux/subst-default-vals.sh \
   build-aux/subst-f77-isnan-macro.sh \
   build-aux/subst-script-vals.sh
 
 $(GEN_CONFIG_SHELL) : %.sh : %.in.sh config.status
 	$(AM_V_GEN)$(SHELL) config.status $@-tmp $@
 
+GEN_CONFIG_INC = \
+  oct-conf-post.h
+
+$(GEN_CONFIG_INC) : %.h : %.in.h config.status
+	$(AM_V_GEN)$(SHELL) config.status $@-tmp $@
+
 AUTOCONF_SUBST_VARS = @AUTOCONF_SUBST_VARS@
 ALL_SUBST_VARS = \
   $(AUTOCONF_SUBST_VARS) \
   abs_top_builddir \
   abs_top_srcdir \
   api_version \
   version
 
@@ -179,16 +185,17 @@ endif
 if AMCOND_HAVE_ICON_TOOLS
 icons-dist-hook:
 else
 icons-dist-hook:
 	@echo "Packaging distribution requires icotool and rsvg-convert." ; exit 1;
 endif
 
 BUILT_SOURCES += \
+  oct-conf-post.h \
   octave-config.h \
   run-octave \
   $(DIRSTAMP_FILES)
 
 if AMCOND_HAVE_BROKEN_STL_ALGO_H
   BUILT_SOURCES += bits/stl_algo.h
 else
   BUILT_SOURCES += nonexistent-file
